SPECS/mariner-rpm-macros/macros

```diff
@@ -69,19 +69,26 @@
 %_use_internal_dependency_generator 1
 %_build_id_links alldebug
 
+# Standard brp-macro naming:
+# convert all '-' in basename to '_', add two leading underscores.
+%__brp_compress %{_rpmconfigdir}/brp-compress %{?_prefix}
+%__brp_strip %{_rpmconfigdir}/brp-strip %{__strip}
+%__brp_strip_comment_note %{_rpmconfigdir}/brp-strip-comment-note %{__strip} %{__objdump}
+%__brp_strip_static_archive %{_rpmconfigdir}/brp-strip-static-archive %{__strip}
+%__brp_elfperms %{_rpmconfigdir}/brp-elfperms
+%__brp_remove_la_files %{_rpmconfigdir}/brp-remove-la-files
+
 %__os_install_post_leave_signatures    \
-    %{_rpmconfigdir}/brp-compress \
-    %{_rpmconfigdir}/brp-strip-comment-note %{__strip} %{__objdump} \
-    %{_rpmconfigdir}/brp-strip-static-archive %{__strip} \
+    %{?__brp_compress} \
+    %{?__brp_strip_comment_note} \
+    %{?__brp_strip_static_archive} \
     %{?py_auto_byte_compile:%{?__brp_python_bytecompile}} \
     find %{buildroot} -name "*.pc" | xargs -I{} sed -i -e 's@-Wl,-dT,%{_topdir}/BUILD/module_info.ld@ @' {} \
     %{nil}
 
 %__os_install_post    \
     %{__os_install_post_leave_signatures} \
-    %{_rpmconfigdir}/brp-strip %{__strip} \
-    %{_rpmconfigdir}/brp-strip-debug-symbols %{__strip} \
-    %{_rpmconfigdir}/brp-strip-unneeded %{__strip} \
```

Comment:
note: removed brp-strip-static-archive due to errors like the below seen in toolchain build:
```
error: File not found: /usr/src/mariner/BUILDROOT/cracklib-2.9.7-5.cm2.x86_64/usr/lib/libcrack.la

" error: File not found: /usr/src/mariner/BUILDROOT/gcc-11.2.0-4.cm2.x86_64/usr/lib64/libstdc++.la 
" error: File not found: /usr/src/mariner/BUILDROOT/gcc-11.2.0-4.cm2.x86_64/usr/lib64/libsupc++.la
```

Reply:
upstream __os_install_post for reference: https://github.com/rpm-software-management/rpm/blob/8ec9cfc81c3f8326a1c7ee5b795ef517cd48e6b4/platform.in#L64

Reply:
I think you meant `brp-remove-la-files`? Reading up on the original [PR](https://github.com/rpm-software-management/rpm/pull/1674) it looks like the upstream thinking was that some components would need to opt-out of this behavior -- perhaps such as the ones you hit errors in -- possibly by redefining the macro wrapping this script to no-op it? I think I'd agree this doesn't need to be done now; do you think it would be worth a follow-up later?

Reply:
Yes, I did mean to say `__brp_remove_la_files`. Seems worthy of follow up, but a quick grep of our core specs shows 200+ packages that perform something along the lines of `find %{buildroot} -type f -name "*.la" -delete -print` already.